examples/tutorial/index.js

```diff
@@ -0,0 +1,306 @@
+const url = require('url');
+const util = require('util');
+const { Vimeo } = require('vimeo');
+const Youtubedl = require('youtube-dl');
+
+Youtubedl.getInfo = util.promisify(Youtubedl.getInfo);
+
+const feedURI = 'https://creativecommons.org/blog/feed/';
+
+const vimeoClientID = '(fill in client ID here)';
+const vimeoClientSecret = '(fill in client secret here)';
+
+const ensureVimeoClient = (function () {
+    let vimeo;
+    return async function _ensureVimeoClient () {
+        if (vimeo) {
+            return vimeo;
+        }
+
+        vimeo = new Vimeo(vimeoClientID, vimeoClientSecret);
+        vimeo.generateClientCredentials =
+            util.promisify(vimeo.generateClientCredentials.bind(vimeo));
+        vimeo.request = util.promisify(vimeo.request.bind(vimeo));
+
+        const { access_token: accessToken } =
+            await vimeo.generateClientCredentials(['public']);
+        vimeo.setAccessToken(accessToken);
+        return vimeo;
+    };
+})();
+
+async function getVideoInfo (vimeoID) {
+    const vimeo = await ensureVimeoClient();
+    const fields = ['description', 'license', 'link', 'name',
+                    'modified_time', 'pictures', 'privacy', 'release_time', 'tags'];
+    return vimeo.request({
+        method: 'GET',
+        path: `/videos/${vimeoID}?fields=${fields.join(',')}`,
+    });
+}
+
+function licenseFromVimeoLicense (license) {
+    switch (license) {
+    case 'by':
+        return 'CC BY 3.0';
+    case 'by-nc':
+        return 'CC BY-NC 3.0';
+    default:
+        console.warn(`Unknown license ${license}`);
+        return null;
+    }
+}
+
+function tagsFromVimeoTags (tags) {
+    return tags.map(({ name }) => name);
+}
+
+function scoreByLength (fnode) {
+    let length = Futils.inlineTextLength(fnode.element) * 2;
+    if (Number.isNaN(length)) {
+        // Penalize empty nodes
+        length = 0;
+    }
+    return {
+        score: length,
+        note: { length },
+    };
+}
+
+function byInverseLinkDensity (fnode) {
+    const linkDensity = Futils.linkDensity(fnode,
+                                           fnode.noteFor('paragraphish').length);
+    if (Number.isNaN(linkDensity)) {
+        return 1;
+    }
+    return (1 - linkDensity) * 1.5;
+}
+
+function scoreByImageSize (fnode) {
+    const img = fnode.element.querySelector('img');
+    const width = img.getAttribute('width');
+    const height = img.getAttribute('height');
+    let length = Futils.inlineTextLength(fnode.element) * 2;
+    if (Number.isNaN(length)) {
+        // Don't penalize empty captions
+        length = 1;
+    }
+    return {
+        score: width && height ? width * height / 100 : 100,
+        note: { length },
+    };
+}
+
+const hasAncestor = (tagName, scoreIfHas) => fnode => {
+    const lowerTag = tagName.toLowerCase();
+    for (let element = fnode.element, parent;
+        (parent = element.parentNode) !== null &&
+         parent.nodeType === parent.ELEMENT_NODE;
+        element = parent) {
```

Comment:
Man, isn't there a way to simplify this `for`? :P

Reply:
Heh, I copy/pasted from the original. Any ideas?

Reply:
Maybe something like:

```
const hasAncestor = (tagName, scoreIfHas) => fnode => {
    const lowerTag = tagName.toLowerCase();

    let element = fnode.element;

    while (element.parentNode !== null && element.parentNode.nodeType === parent.ELEMENT_NODE) {
        if (element.tagName.toLowerCase() === lowerTag) {
            return scoreIfHas;
        }
        element = element.parentNode;
    }

    return 1;
};
```

Reply:
+1